frontend/public/components/utils/tile-view-page.jsx

```diff
@@ -0,0 +1,686 @@
+  }
+
+  setURLParams(params);
+};
+
+const getActiveValuesFromURL = (availableFilters, filterGroups) => {
+  const searchParams = new URLSearchParams(window.location.search);
+  const categoryParam = searchParams.get(CATEGORY_URL_PARAM);
+  const keywordFilter = searchParams.get(KEYWORD_URL_PARAM);
+
+  const selectedCategoryId = categoryParam || 'all';
+
+  const groupFilters = {};
+
+  _.each(filterGroups, filterGroup => {
+    const groupFilterParam = searchParams.get(filterGroup);
+    if (!groupFilterParam) {
+      return;
+    }
+
+    try {
+      _.set(groupFilters, [filterGroup], JSON.parse(groupFilterParam));
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.warn('could not update filters from url params: could not parse search params', e);
+    }
+  });
+
+  const activeFilters = getActiveFilters(keywordFilter, groupFilters, availableFilters);
+
+  return {selectedCategoryId, activeFilters};
+};
+
+const getFilterSearchParam = groupFilter => {
+  const activeValues = _.reduce(_.keys(groupFilter), (result, typeKey) => {
+    return groupFilter[typeKey].active ? result.concat(typeKey) : result;
+  }, []);
+
+  return _.isEmpty(activeValues) ? '' : JSON.stringify(activeValues);
+};
+
+const defaultFilters = {
+  keyword: {
+    value: '',
+    active: false,
+  },
+};
+
+export class TileViewPage extends React.Component {
+  constructor(props) {
+    super(props);
+    const {items, itemsSorter, filterGroups, getAvailableCategories, getAvailableFilters} = this.props;
+
+    // Filters are populated based on white list of item properties
+    const availableFilters = getAvailableFilters(defaultFilters, items, filterGroups);
+
+    const categories = getAvailableCategories(items);
+
+    this.state = {
+      categories: categorizeItems(items, itemsSorter, categories),
+      selectedCategoryId: 'all',
+      availableFilters,
+      activeFilters: availableFilters,
+      filterCounts: null,
+    };
+  }
+
+  componentDidMount() {
+    const {filterGroups} = this.props;
+    const {categories, availableFilters} = this.state;
+
+    const activeValues = getActiveValuesFromURL(availableFilters, filterGroups);
+
+    this.setState({...this.getUpdatedState(categories, activeValues.selectedCategoryId, activeValues.activeFilters) });
+  }
+
+  componentWillUnmount() {
+    this.unmounted = true;
+  }
+
+  componentDidUpdate(prevProps) {
+    const { activeFilters, selectedCategoryId } = this.state;
+    const { items, itemsSorter, filterGroups, getAvailableCategories, getAvailableFilters } = this.props;
+
+    if (!_.isEqual(items, prevProps.items)) {
+      const availableFilters = getAvailableFilters(defaultFilters, items, filterGroups);
+      const marketplaceCategories = getAvailableCategories(items);
+      const categories = categorizeItems(items, itemsSorter, marketplaceCategories);
+      this.setState({availableFilters, ...this.getUpdatedState(categories, selectedCategoryId, activeFilters)});
```

Comment:
Does `activeFilters` get updated to match the potential changes to `availableFilters`?

Reply:
Looking into that @SamiSousa, they ought to be...

Reply:
Updated to correctly update the activeFilters, no longer a need to save availableFilters in state.